api/client/samples/analogous_years/lib/get_transform_data.py

```diff
@@ -0,0 +1,140 @@
+"""
+This library contains utilities for
+1. retrieving data series from gro api
+2. extracting and transforming relevant part of the data series
+"""
+
+from dateutil.parser import parse
+from dateutil.relativedelta import relativedelta
+import pandas as pd
+
+
+def get_data(client, metric_id, item_id, region_id, source_id, frequency_id, start_date):
+    """
+    :param client: GroClient
+    :param metric_id: Gro-metric
+    :param item_id: Gro-item associated to the metric
+    :param region_id: Gro-region
+    :param source_id: Gro-source
+    :param frequency_id: Gro-frequency
+    :param start_date: start-date of the Gro data series
+    :return: A dataframe with an 'end_date' and 'value' column
+    """
+    data = client.get_data_points(**{'metric_id': metric_id,
+                                     'item_id': item_id,
+                                     'region_id': region_id,
+                                     'source_id': source_id,
+                                     'frequency_id': frequency_id,
+                                     'start_date': start_date})
+    data = pd.DataFrame(data)
```

Comment:
use

```
for series in the series you wnat:
   client.add_single_data_series(** series)

client.get_df()
```
this will give you a dataframe that contains the

Reply:
This does not work as you proposed. I envisioned that users will copy the data series values from the web app and then paste it in their workflow. The method you described will take input a bit differently.